src/main/java/teammates/common/datatransfer/questions/FeedbackQuestionDetails.java

```diff
@@ -92,18 +99,32 @@ public String getCsvDetailedResponsesRow(FeedbackSessionResultsBundle fsrBundle,
         String recipientTeamName = fsrBundle.getTeamNameForEmail(feedbackResponseAttributes.recipient);
         String recipientEmail = fsrBundle.getDisplayableEmailRecipient(feedbackResponseAttributes);
 
-        return SanitizationHelper.sanitizeForCsv(StringHelper.removeExtraSpace(giverTeamName))
+        StringBuilder detailedResponseRow = new StringBuilder(1000);
+        String detailedResponseRowString = SanitizationHelper.sanitizeForCsv(StringHelper.removeExtraSpace(giverTeamName))
                 + "," + SanitizationHelper.sanitizeForCsv(StringHelper.removeExtraSpace(giverFullName))
                 + "," + SanitizationHelper.sanitizeForCsv(StringHelper.removeExtraSpace(giverLastName))
                 + "," + SanitizationHelper.sanitizeForCsv(StringHelper.removeExtraSpace(giverEmail))
                 + "," + SanitizationHelper.sanitizeForCsv(StringHelper.removeExtraSpace(recipientTeamName))
                 + "," + SanitizationHelper.sanitizeForCsv(StringHelper.removeExtraSpace(recipientFullName))
                 + "," + SanitizationHelper.sanitizeForCsv(StringHelper.removeExtraSpace(recipientLastName))
                 + "," + SanitizationHelper.sanitizeForCsv(StringHelper.removeExtraSpace(recipientEmail))
-                + "," + fsrBundle.getResponseAnswerCsv(feedbackResponseAttributes, question)
-                + (hasCommentsForResponses
-                        ? fsrBundle.getCsvDetailedFeedbackResponseCommentsRow(feedbackResponseAttributes) : "")
-                + System.lineSeparator();
+                + "," + fsrBundle.getResponseAnswerCsv(feedbackResponseAttributes, question);
+        detailedResponseRow.append(detailedResponseRowString);
+        //Appends feedback participant comments if allowed
+        if (isFeedbackParticipantCommentsOnResponsesAllowed()) {
+            String feedbackParticipantComment =
+                    fsrBundle.getCsvDetailedFeedbackParticipantCommentOnResponse(feedbackResponseAttributes);
+            if (!feedbackParticipantComment.isEmpty()) {
+                detailedResponseRow.append(',').append(feedbackParticipantComment);
```

Comment:
Will this result in malformatted CSV?

Some response doesn't have participants' comment but has instructors' comment. If you don't append the comma, the instructors' comment will fall into the column for participants' comments?